pkg/mac/build-static-library.sh

```diff
@@ -18,182 +18,48 @@
-        lipo libprotobuf_universal.a -thin ${ARCH} -output libprotobuf.a
-      popd
-      touch .done
-    popd
-else
-    echo "Using cached Protobuf $PROTOBUF_VERSION"
-fi
-
-if [ ! -f zstd-${ZSTD_VERSION}/.done ]; then
-    echo "Building ZStd $ZSTD_VERSION"
-    curl -O -L https://github.com/facebook/zstd/releases/download/v${ZSTD_VERSION}/zstd-${ZSTD_VERSION}.tar.gz
-    tar zxf zstd-${ZSTD_VERSION}.tar.gz
-    pushd zstd-${ZSTD_VERSION}
-      CFLAGS="-fPIC -O3 -arch ${ARCH} -mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET}" PREFIX=$PREFIX \
-            make -j16 -C lib install-static install-includes
-      touch .done
-    popd
-else
-    echo "Using cached ZStd $ZSTD_VERSION"
-fi
-
-if [ ! -f snappy-${SNAPPY_VERSION}/.done ]; then
-    echo "Building Snappy $SNAPPY_VERSION"
-    curl -O -L https://github.com/google/snappy/archive/refs/tags/${SNAPPY_VERSION}.tar.gz
-    tar zxf ${SNAPPY_VERSION}.tar.gz
-    pushd snappy-${SNAPPY_VERSION}
-      # Without this patch, snappy 1.10 will report a sign-compare error, which cannot be suppressed with the -Wno-sign-compare option in CI
-      curl -O -L https://raw.githubusercontent.com/microsoft/vcpkg/2024.02.14/ports/snappy/no-werror.patch
-      patch <no-werror.patch
-      CXXFLAGS="-fPIC -O3 -arch ${ARCH} -mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET}" \
-          cmake . -DCMAKE_INSTALL_PREFIX=$PREFIX -DSNAPPY_BUILD_TESTS=OFF -DSNAPPY_BUILD_BENCHMARKS=OFF
-      make -j16
-      make install
-      touch .done
-    popd
-else
-    echo "Using cached Snappy $SNAPPY_VERSION"
-fi
-
-if [ ! -f curl-${CURL_VERSION}/.done ]; then
-    echo "Building LibCurl $CURL_VERSION"
-    CURL_VERSION_=${CURL_VERSION//./_}
-    curl -O -L https://github.com/curl/curl/releases/download/curl-${CURL_VERSION_}/curl-${CURL_VERSION}.tar.gz
-    tar zxf curl-${CURL_VERSION}.tar.gz
-    pushd curl-${CURL_VERSION}
-      # Force the compiler to find the OpenSSL headers instead of the headers in the system path like /usr/local/include/openssl.
-      cp -rf $PREFIX/include/openssl include/
-      CFLAGS="-I$PREFIX/include -fPIC -arch ${ARCH} -mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET}" \
-            ./configure --with-ssl=$PREFIX \
-              --without-nghttp2 \
-              --without-libidn2 \
-              --disable-ldap \
-              --without-brotli \
-              --without-secure-transport \
-              --without-librtmp \
-              --disable-ipv6 \
-              --without-libpsl \
-              --host=$ARCH-apple-darwin \
-              --prefix=$PREFIX
-      make -j16 install
-      touch .done
-    popd
-else
-    echo "Using cached LibCurl $CURL_VERSION"
-fi
-
-popd # pkg/mac
-cd ../../ # project root
-
-cmake -B build-static -DCMAKE_OSX_DEPLOYMENT_TARGET=$MACOSX_DEPLOYMENT_TARGET \
-    -DLINK_STATIC=ON \
+# Use a forked version of vcpkg to support building libcurl with IPv6 disabled
+rm -rf vcpkg
+git clone https://github.com/BewareMyPower/vcpkg.git -b curl-8.4.0-osx-patch
```

Comment:
How about using a patch file for the commit https://github.com/BewareMyPower/vcpkg/commit/b03173a5d0cbd2f7c3c5e85a576e98fba3560835 so that we don't need to introduce the personal repo?

Reply:
Okay, I think we can restore the solution in https://github.com/apache/pulsar-client-cpp/pull/465/commits/382c9d9a5fcc3212987685d416b06983e9af82c6